Signal/src/ViewControllers/SignalsViewController.m

```diff
@@ -861,20 +862,28 @@ - (YapDatabaseConnection *)uiDatabaseConnection {
 }
 
 - (void)yapDatabaseModified:(NSNotification *)notification {
+    if (!self.shouldObserveDBModifications) {
+        return;
+    }
+
     NSArray *notifications  = [self.uiDatabaseConnection beginLongLivedReadTransaction];
-    NSArray *sectionChanges = nil;
-    NSArray *rowChanges     = nil;
+
+    if (![[self.uiDatabaseConnection ext:TSThreadDatabaseViewExtensionName] hasChangesForGroup:self.currentGrouping
```

Comment:
Although `hasChangesForGroup:` filters the modifications by group, the most important filtering it does is to ignore all database changes that don't affect `TSThreadDatabaseViewExtensionName`.  E.g. writing a received message to the durable queue, saving a prekey change, etc.

> Will this solve the problem of requesting contact access when receiving your first message while the app is in the background?

It might.  We should test that.

> I wonder if there's an edge case where we won't properly request contact access if you are viewing your archive when you receive your first message.

...or if the app was in the background when you received your first message.  Sounds like we should add a check for that in the new `resetMappings` method.  I'll do so.